blog/content/posts/2025-12-19-mcp-transport-future.md

```diff
@@ -0,0 +1,92 @@
+
+To address this, we're looking at designing server requests and responses to work similarly to chat APIs. The server returns the elicitation request as usual, and the client returns both the request _and_ response together. This allows the server to reconstruct the necessary state purely from the returned message, avoiding long-running state management between nodes and potentially eliminating the need for back-end storage entirely.
+
+### Update Notifications and Subscriptions
+
+MCP is dynamic by design - [tools](https://modelcontextprotocol.io/specification/2025-11-25/server/tools), [prompts](https://modelcontextprotocol.io/specification/2025-11-25/server/prompts), and [resources](https://modelcontextprotocol.io/specification/2025-11-25/server/resources) can change during operation. Today, servers send `ListChangedNotification` messages to clients as a hint to invalidate their caches.
+
+We're exploring replacing the general-purpose `GET` stream with explicit subscription streams. Clients would open dedicated streams for specific items they want to monitor, with support for multiple concurrent subscriptions. If a stream is interrupted, the client simply restarts it with no complex resumption logic.
+
+To make notifications truly optional - an optimization rather than a requirement - we're considering adding Time-To-Live (TTL) values and version identifiers (such as [ETags](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/ETag)) to data. This would let clients make intelligent caching decisions independently of the notification stream, significantly improving reliability.
+
+### JSON-RPC Envelopes
+
+MCP uses JSON-RPC for all message envelopes, including method names and parameters. As we optimize for HTTP deployments, a common question is whether routing information should be more accessible to the underlying MCP server infrastructure.
+
+While we're keeping JSON-RPC as the message format, we're exploring ways to expose routing-critical information (such as the RPC method or tool name) via standard HTTP paths or headers. This would allow load balancers and API gateways to route traffic without parsing JSON bodies.
+
+### Pluggable Transports
+
+The MCP Specification already supports [Custom Transports](https://modelcontextprotocol.io/specification/2025-11-25/basic/transports#custom-transports), allowing integrators to deploy alternatives that fit their specific needs.
+
+STDIO and Streamable HTTP are designated as **Standard** transports, with guaranteed SDK support and ecosystem-wide interoperability by default.
+
+Rather than adding new standard transport types, which would fragment connectivity options, we're focusing on making Custom Transports easier to implement by improving how they integrate with existing protocol SDKs.
+
+## Summary
```

Comment:
I think somewhere in here we should emphasize the principle around minimal breaking changes. This will also help give the community confidence that we aren't going to be introducing hugely disruptive changes this year. 

Something like "we know Transport & Protocol changes can be disruptive to developers, and we will be making these changes with that in mind.  We are prioritizing backwards compatibility and only implementing breaking changes where strictly necessary to critical use cases."